v0/destinations/adobe_analytics/transform.js

```diff
@@ -288,10 +288,14 @@ const processTrackEvent = (
         productMerchEventToAdobeEvent[event.toLowerCase()] &&
         productMerchProperties
       ) {
+        adobeProdEvent = adobeProdEvent.toString().split(",");
         productMerchProperties.forEach(rudderProp => {
-          if (rudderProp.productMerchProperties.startsWith("products.")) {
+          if (
+            rudderProp.productMerchProperties.startsWith("products.") &&
+            isSingleProdEvent === false
+          ) {
             const key = rudderProp.productMerchProperties.split(".");
-            const v = get(properties, key[1]);
+            const v = get(value, key[1]);
             if (isDefinedAndNotNull(v)) {
               Object.keys(adobeProdEvent).forEach(val => {
```

Comment:
It's an array, `adobeProdEvent.forEach` should work.

Reply:
addressed